skpro/distributions/qpd_empirical.py

```diff
@@ -10,11 +10,40 @@
 
 
 class QPD_Empirical(Empirical):
-    """Empirical quantile parametrized distribution.
-
-    This distribution is parameterized by a set of quantile points.
-
-    todo: add docstr
+    r"""Empirical quantile parametrized distribution.
+
+    This distribution is parameterized by a set of quantile points and quantiles,
+    quantiles :math:`q_1, q_2, \dots, q_N`
+    at quantile points :math:`p_1, p_2, \dots, p_N`,
+    with :math:`0 \le p_1 < p_2 < \dots < p_N \le 1`.
+
+    It represents a distribution with piecewise constant CDF and quantile function,
+    the unique distribution satisfying:
+
+    * the support is :math:`[q_1, q_N]`
+    * for any quantile point :math:`p \in [p_1, p_N]`, it holds that
+      :math:`\mbox{ppf}(p)` = :math:`\mbox{ppf}(p_i)`,
+      where :math:`i` is the index minimizing :math:`|p_i - p|`,
+      in all cases where this minimizer is unique.
+
+    In vernacular terms, the quantile function agrees with the quantiles prescribed by
+    :math:`q_i` at the quantile points :math:`p_i`, and for other quantile points
+    agrees with the value at the nearest quantile point.
+
+    In explicit terms, the distribution is an empirical distribution (sum-of-diracs),
+    supported at the quantiles :math:`q_1, q_2, \dots, q_N`,
+    with weights :math:`w_1, w_2, \dots, w_N`
+    such that :math:`w_i = (p_{i+1} - p_{i-1})/2` for :math:`1 = 1, \dots, N`,
+    where we define :math:`p_0 = -p_1` and :math:`p_{N+1} = 2 - p_N`.
```

Comment:
Just one question:
you say: `such that w_i = (p_{i+1} - p_{i-1})/2  for 1 = 1, \dots, N ` and then you define `p_0` and `p_{N+1}`. Why you define them? How are these two values playing a role here?

Reply:
Ah sorry now I see ;-)

thanks for the explanation. I didn't check that there it was a `p_{i-1})` and we have to define `p_0` (same for `p_{i+1}`

Reply:
the $w_i$ should sum to 1, hopefully